alfresco-solrapi-client/alfresco-solrapi-client-spring/src/main/java/eu/xenit/alfresco/solrapi/client/spring/SolrApiSpringClient.java

```diff
@@ -72,22 +76,51 @@ private static RestTemplate buildRestTemplate(ClientHttpRequestFactory requestFa
         return client;
     }
 
+
     @Override
-    public AclChangeSets getAclChangeSets(Long fromCommitTime, Long minAclChangeSetId, Long toCommitTime,
-            Long maxAclChangeSetId, int maxResults) {
-        throw new UnsupportedOperationException();
+    public List<AclChangeSet> getAclChangeSets(Long fromId, Long fromTime, int maxResults)
+    {
+        UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromHttpUrl(url).path("/aclchangesets");
+        conditionalQueryParam(uriBuilder, "fromId", fromId, Objects::nonNull);
+        conditionalQueryParam(uriBuilder, "fromTime", fromTime, Objects::nonNull);
+        conditionalQueryParam(uriBuilder, "maxResults", maxResults, val ->
+                val != null && val != Integer.valueOf(0) && val != Integer.valueOf(Integer.MAX_VALUE));
```

Comment:
checking an int for not being `null` will always succeed

Reply:
Correct pls in getTransactions